lms/static/js/spec/student_profile/learner_profile_view_spec.js

```diff
@@ -0,0 +1,176 @@
+            });
+
+            it("shows loading error correctly", function() {
+
+                var learnerProfileView = createLearnerProfileView(false, 'all_users');
+
+                Helpers.expectLoadingIndicatorIsVisible(learnerProfileView, true);
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+
+                learnerProfileView.render();
+                learnerProfileView.showLoadingError();
+
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, true);
+            });
+
+            it("renders all fields as expected for self with full access", function() {
+
+                var learnerProfileView = createLearnerProfileView(true, 'all_users', true);
+
+                Helpers.expectLoadingIndicatorIsVisible(learnerProfileView, true);
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+
+                learnerProfileView.render();
+
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+                Helpers.expectProfileSectionsAndFieldsToBeRendered(learnerProfileView);
+            });
+
+            it("renders all fields as expected for self with limited access", function() {
+
+                var learnerProfileView = createLearnerProfileView(true, 'private', false);
+
+                Helpers.expectLoadingIndicatorIsVisible(learnerProfileView, true);
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+
+                learnerProfileView.render();
+
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+                Helpers.expectLimitedProfileSectionsAndFieldsToBeRendered(learnerProfileView);
+            });
+
+            it("renders the fields as expected for others with full access", function() {
+
+                var learnerProfileView = createLearnerProfileView(false, 'all_users', true);
+
+                Helpers.expectLoadingIndicatorIsVisible(learnerProfileView, true);
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+
+                learnerProfileView.render();
+
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+                Helpers.expectProfileSectionsAndFieldsToBeRendered(learnerProfileView, true)
+            });
+
+            it("renders the fields as expected for others with limited access", function() {
+
+                var learnerProfileView = createLearnerProfileView(false, 'private', false);
+
+                Helpers.expectLoadingIndicatorIsVisible(learnerProfileView, true);
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+
+                learnerProfileView.render();
+
+                Helpers.expectLoadingErrorIsVisible(learnerProfileView, false);
+                Helpers.expectLimitedProfileSectionsAndFieldsToBeRendered(learnerProfileView, true);
+            });
+        });
```

Comment:
Please add tests for editing each of the editable fields with both successful responses and for error responses.